src/main/java/org/mule/extension/ftp/internal/operation/BaseFileSystemOperations.java

```diff
@@ -0,0 +1,343 @@
+   * @param sourcePath              the path to the file to be copied
+   * @param targetPath              the target directory
+   * @param createParentDirectories whether or not to attempt creating any parent directories which don't exists.
+   * @param overwrite               whether or not overwrite the file if the target destination already exists.
+   * @param renameTo                the new file name, {@code null} if the file doesn't need to be renamed
+   * @throws IllegalArgumentException if an illegal combination of arguments is supplied
+   */
+  protected void doMove(FileConnectorConfig config, FileSystem fileSystem, String sourcePath,
+                        String targetPath, boolean createParentDirectories, boolean overwrite, String renameTo) {
+    fileSystem.changeToBaseDir();
+    validatePath(targetPath, "target path");
+    validatePath(sourcePath, "source path");
+    fileSystem.move(config, sourcePath, targetPath, overwrite, createParentDirectories, renameTo);
+  }
+
+  /**
+   * Deletes the file pointed by {@code path}, provided that it's not locked
+   *
+   * @param fileSystem a reference to the host {@link FileSystem}
+   * @param path       the path to the file to be deleted
+   * @throws IllegalArgumentException if {@code filePath} doesn't exist or is locked
+   */
+  protected void doDelete(FileSystem fileSystem, @Optional String path) {
+    fileSystem.changeToBaseDir();
+    fileSystem.delete(path);
+  }
+
+  /**
+   * Renames the file pointed by {@code path} to the name provided on the {@code to} parameter
+   * <p>
+   * {@code to} argument should not contain any path separator. {@link IllegalArgumentException} will be thrown if this
+   * precondition is not honored.
+   *  @param fileSystem a reference to the host {@link FileSystem}
+   * @param path       the path to the file to be renamed
+   * @param to         the file's new name
+   * @param overwrite  whether or not overwrite the file if the target destination already exists.
+   */
+  protected void doRename(@Connection FileSystem fileSystem, @Optional String path,
+                          @DisplayName("New Name") String to, @Optional(defaultValue = "false") boolean overwrite) {
+    if (get(to).getNameCount() != 1) {
+      throw new IllegalPathException(
+                                     format("'to' parameter of rename operation should not contain any file separator character but '%s' was received",
+                                            to));
+    }
+
+    fileSystem.changeToBaseDir();
+    fileSystem.rename(path, to, overwrite);
+  }
+
+  /**
+   * Creates a new directory on {@code directoryPath}
+   *
+   * @param fileSystem    a reference to the host {@link FileSystem}
+   * @param directoryPath the new directory's name
+   */
+
+  protected void doCreateDirectory(@Connection FileSystem fileSystem, String directoryPath) {
+    validatePath(directoryPath, "directory path");
+    fileSystem.changeToBaseDir();
```

Comment:
paths can be absolute

Reply:
this is an existing call and internally checking if it's absolute path then it's not doing anything so should be good here.